ssl/bio_ssl.c

```diff
@@ -382,14 +382,6 @@ static long ssl_ctrl(BIO *b, int cmd, long num, void *ptr)
     case BIO_CTRL_SET_CALLBACK:
         ret = 0; /* use callback ctrl */
         break;
-    case BIO_CTRL_GET_CALLBACK:
-        {
-            void (**fptr) (const SSL *xssl, int type, int val);
-
-            fptr = (void (**)(const SSL *xssl, int type, int val))ptr;
-            *fptr = SSL_get_info_callback(ssl);
-        }
-        break;
```

Comment:
I don't follow. It appears to be replacing assignment to ssl->info_callback with assignment to ssl->rbio->info_callback. Is this right/intentional? ssl->info_callback is used e.g. in ssl3_dispatch_alert...

Reply:
I had no better idea, fact is SSL_get_info_callback will invoke the callback with
SSL* instead of BIO *

Reply:
One thing I considered here was adding some glue code that translates the SSL* to the BIO*,
but I had no idea how to accomplish this,

Reply:
The invokation sounds buggy...